cmd/xl-storage-format-v2.go

```diff
@@ -1080,12 +1086,23 @@ func (z *xlMetaV2) DeleteVersion(fi FileInfo) (string, bool, error) {
 		}
 	}
 
-	findDataDir := func(dataDir [16]byte, versions []xlMetaV2Version) int {
+	findDataDir := func(v2 *xlMetaV2Object, versions []xlMetaV2Version, inlineVersions []string) int {
```

Comment:
We only need to check if `v2`'s contents are inlined. So it's cheaper to use `z.data.find(uuid.UUID(v2.VersionID).String()) != nil` to know if its contents are inlined.

Other object versions will share data directory if and only if they were created from a CopyObject (metadata only) operation. This implies their content and size are equal (obviously) and by extension they would either all be inlined or not. So, if `v2` is inlined then other versions copied from must be too. Otherwise, other versions aren't inlined and will share the data directory.

Reply:
So, the only change `findDataDir` needs is,
```go
if z.data.find(uuid.UUID(v2.VersionID).String()) != nil {
    return 0
}
```